ansible/roles/helios_broker_setup/tasks/main.yml

```diff
@@ -0,0 +1,142 @@
+      dockerhub_org: "{{ helios_dockerhub_org }}"
+      openshift_pass: "{{ cluster_user_password }}"
+      openshift_user: "{{ cluster_user }}"
+      openshift_target: "{{ cluster_url }}"
+
+  - debug: var=dockerhub_org
+
+  - name: check if the {{ helios_asb_project }} project exists
+    shell: "{{ oc_cmd }} get project --no-headers=true | awk '{ print $1 }' | grep -E '^{{ helios_asb_project }}( |$)' | cat"
+    register: project
+
+  - name: retrieve cert for cluster
+    shell: "{{ oc_cmd}} get secret --no-headers=true -n kube-service-catalog | grep -m 1 service-catalog-apiserver-token | {{ oc_cmd}} get secret $(awk '{ print $1 }') -n kube-service-catalog -o yaml | grep service-ca.crt | awk '{ print $2 }' | cat"
+    register: cluster_ca_cert
+
+  - debug:
+      msg: "Service Broker must have a CA Bundle set for the Helios Service Broker and Service Catalog to be able to communicate"
+    when: cluster_ca_cert.stdout == "" and openshift_client_version != "3.6"
+
+  - name: Create a new project for the {{ helios_asb_project }}
+    shell: "{{ oc_cmd }} new-project {{ helios_asb_project }}"
+    register: new_helios_asb_project
+    when: project.stdout.find( helios_asb_project ) == -1
+
+  - name: Update Broker Kind and Auth when using coalmine
+    set_fact:
+      broker_enable_basic_auth: true
+    when: openshift_client_version == '3.6'
+  #####
+  # Remember that 'rcm' builds are using an older branch of the broker
+  # Some features in the deployment template may not be available, such as
+  #     REGISTRY_NAME
+  # Goal is to build up a common command, then modify it as needed for newer
+  # features.
+  #####
+  - name: Create command line for processing template
+    set_fact:
+      cmd_process_asb_template: >-
+        {{ oc_cmd }}
+        process
+        -f {{ helios_local_target_asb_template }}
+        -n {{ helios_asb_project }}
+        -p BROKER_IMAGE={{ helios_broker_image }}
+        -p ETCD_IMAGE={{ etcd_image }}
+        -p ETCD_PATH={{ etcd_path }}
+        -p REGISTRY_TYPE={{ broker_registry_type }}
+        -p REGISTRY_URL={{ broker_registry_url }}
+        -p DEV_BROKER={{ broker_dev_broker }}
+        -p DOCKERHUB_ORG='{{ helios_dockerhub_org }}'
+        -p DOCKERHUB_PASS='{{ dockerhub_user_password }}'
+        -p DOCKERHUB_USER='{{ dockerhub_user_name }}'
+        -p LAUNCH_APB_ON_BIND={{ broker_launch_apb_on_bind }}
+        -p OUTPUT_REQUEST={{ broker_output_request }}
+        -p RECOVERY={{ broker_recovery }}
+
+  - name: Modify command if running a non-rcm build to use newer features
+    set_fact:
+      cmd_process_asb_template: >-
+        {{ cmd_process_asb_template }}
+        -p ROUTING_SUFFIX={{ openshift_routing_suffix }}
+        -p REGISTRY_NAME={{ broker_registry_name }}
+        -p ENABLE_BASIC_AUTH={{ broker_enable_basic_auth }}
+        -p REFRESH_INTERVAL={{ broker_bootstrap_refresh_interval }}
+        -p TAG={{ apbtag | default('latest') }}
```

Comment:
We will want to prefix this to helios_apbtag

Reply:
will update